### PR TITLE
(MODULES-4750) puppetlabs-mysql : mysqld_version is inconsistent

### PR DESCRIPTION
**Pull request by shawnferry**
This patch redefines the mysqld_version fact to major.minor.patch matching
  the behavior of mysql_value. It also defines mysqld_version_string and
  mysqld_type.

  Previously mysqld_version was stored by the class as mysqld_version_string,
  The mysqld_type and mysqld_version methods parsed mysqd_version_string on
  demand.

  Remote instances may need to override mysqld_version and mysqld_type.

### CODE REVIEW

**Comment by david22swan** [comment]
@shawnferry Please rebase your changes onto the latest branch.
In addition as this PR was opened prior to the implementation of rubocop upon this module you should run it in order to ensure that your changes match the set criteria.
Some simple rubocop commands are included below:

Run command:
bundle exec rubocop

Flags:
-a : run autocorrect
--display-cop-names : add so that each error shows its flag name

Disable Commands:
Disable rubocop over a block off code
   # rubocop:disable Lint/UselessAssignment
        {Code Block}
    # rubocop:enable Lint/UselessAssignment

Disable rubocop for a single line:
    {Code Line} # rubocop:disable RSpec/InstanceVariable

File: lib/facter/mysqld_version.rb [hunk 1]
```diff
@@ -1,5 +1,29 @@
...
+  end
+
+  Facter.add('mysqld_type') do
+    setcode do
+      # find the mysql "dialect" like mariadb / mysql etc.
+      if version_string =~ %r{mariadb}i
+        'mariadb'
+      elsif version_string =~ %r{\s\(percona}i
+        'percona'
+      else
```
Comment (eputnam) [inline lib/facter/mysqld_version.rb]: i'd like to have an elsif for mysql and then maybe else return "unknown"

File: spec/unit/puppet/provider/mysql_user/mysql_spec.rb [hunk 2]
```diff
@@ -2,26 +2,26 @@
...
   # Output of mysqld -V
-  mysql_version_string_hash = {
-    'mysql-5.5' =>
-    {
-      version: '5.5.46',
-      string: '/usr/sbin/mysqld  Ver 5.5.46-log for Linux on x86_64 (MySQL Community Server (GPL))',
-      mysql_type: 'mysql',
-    },
-    'mysql-5.6' =>
+  $mysql_version_string_hash = {
```
Comment (eputnam) [inline spec/unit/puppet/provider/mysql_user/mysql_spec.rb]: what does the $ do here? is this rspec-ese?

**Comment by eputnam** [comment]
Closing this due to inactivity. @shawnferry please feel free to re-open if you get some time to work on it.

**Pull request closed without merging**